### PR TITLE
biom convert finalization

### PR DESCRIPTION
**Pull request by jairideout**
Contains commits in #670.

### CODE REVIEW

**Comment by wasade** [comment]
@jairideout, same comment though, possible to pulldown #661?

**Comment by jairideout** [comment]
Can we get this merged first? I don't see the benefit of pulling in #661 at this point. I need this merged to tackle #283 and #656.

**Comment by wasade** [comment]
The py3 changes either happen in this PR or that one... Guess it's a race
to see which gets merged first :)

**Comment by jairideout** [comment]
I'm currently blocked and this PR is smaller, sooooooo my vote goes on this one

**Comment by wasade** [comment]
well... its not the size of the pr that matters. but, this one _is_ larger by LOC...

File: tests/test_cli/test_table_converter.py [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+
+
+class TableConverterTests(TestCase):
+
+    def setUp(self):
+        """Set up data for use in unit tests."""
+        self.cmd = convert
+        self.output_filepath = tempfile.NamedTemporaryFile().name
+
+        with tempfile.NamedTemporaryFile() as fh:
```
Comment (wasade) [inline tests/test_cli/test_table_converter.py]: can you add `mode='w'`? py3 vomits otherwise

Comment (wasade) [inline tests/test_cli/test_table_converter.py]: ...same for other patterns like this below

File: tests/test_cli/test_table_converter.py [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+                       sample_metadata=self.sample_md1, to_json=True,
+                       table_type='OTU table', process_obs_metadata='naive')
+
+        obs = load_table(self.output_filepath)
+        self.assertEqual(type(obs), Table)
+        self.assertEqual(len(obs.ids()), 9)
+        self.assertEqual(len(obs.ids(axis='observation')), 14)
+        self.assertNotEqual(obs.metadata(), None)
+        self.assertNotEqual(obs.metadata(axis='observation'), None)
+        self.assertEqual(obs.metadata()[obs.index('p2', 'sample')],
```
Comment (wasade) [inline tests/test_cli/test_table_converter.py]: possible to decorate the string literals with 'u'?

File: tests/test_cli/test_table_converter.py [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+    def test_json_to_hdf5_collapsed_samples(self):
+        """Correctly converts json to HDF5 changing the sample metadata"""
+        with biom_open(self.json_collapsed_samples) as f:
+            obs = self.cmd(table=parse_biom_table(f),
+                           output_filepath=self.output_filepath, to_hdf5=True,
+                           collapsed_samples=True)
+        obs = load_table(self.output_filepath)
+        exp = Table(np.array([[0., 1.], [6., 6.], [6., 1.],
+                              [1., 4.], [0., 2.]]),
+                    observation_ids=['GG_OTU_1', 'GG_OTU_2', 'GG_OTU_3',
```
Comment (wasade) [inline tests/test_cli/test_table_converter.py]: same here and below re: "u"

**Comment by wasade** [comment]
@jairideout, I marked what I think will be the primary py3 gotchas. They should be quick to do

File: tests/test_cli/test_table_converter.py (hunk 3 above)
Comment (jairideout) [inline tests/test_cli/test_table_converter.py]: done

File: tests/test_cli/test_table_converter.py (hunk 2 above)
Comment (jairideout) [inline tests/test_cli/test_table_converter.py]: done

File: tests/test_cli/test_table_converter.py (hunk 1 above)
Comment (jairideout) [inline tests/test_cli/test_table_converter.py]: done

**Comment by jairideout** [comment]
@wasade thanks for reviewing, let me know of any other changes

**Comment by wasade** [comment]
looks good, :+1: merge on tests passing

**Pull request merged**